tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_types.ts

```diff
@@ -123,6 +123,88 @@ export interface GraphExecution extends GraphExecutionDigest {
   device_name: string;
 }
 
+/**
+ * Information about an op in a TensorFlow graph.
+ *
+ * Including its enclosing graph, relation with other ops in the graph
+ * (inputs and consumers), and the source-code location (stack trace)
+ * at which the op was created.
+ */
+export interface GraphOpInfo {
+  // Op type (e.g., "MatMul").
+  op_type: string;
+
+  // Op name, i.e., name of the node in the graph, (e.g., "Dense_2/MatMul").
+  op_name: string;
+
+  device_name: string;
+
+  // IDs of the enclosing graphs for this op, from outermost to innermost.
+  graph_ids: string[];
+
+  // Number of symoblic tensors output by the op.
+  num_outputs: number;
+
+  // Debugger-generated IDs for the symbolic output tensor(s) of this op.
+  // For an op without output tensors, this is an empty array.
+  output_tensor_ids: number[];
+
+  // The name of the host on which the op is created.
+  host_name: string;
+
+  // IDs of the frame of the stack trace at which the op is created.
+  stack_frame_ids: string[];
+
+  // Op names and slots of the immediate data input to the op.
+  //`[]` if an op has no data inputs tensors.
+  // This field does *not* track control inputs.
+  // E.g., `[{op_name: "Dense_2/ReadVariableOp_1:0", ouput_slot: 0},
+  //         {op_name: "Input:0", output_slot: 0}]`
+  inputs: GraphOpInputSpec[];
+
+  // Op names and slots of the immediate consumers of the op's output tenors.
+  // `[]` if the op provides no output tensors.
+  // If any of the output tensors of the op has no consumers, the corresponding
+  // element will be `[]`.
+  consumers: GraphOpConsumerSpec[][];
+}
+
+/**
+ * Specificaton of an input tensor to a graph op.
+ */
+export interface GraphOpInputSpec {
+  // Name of the graph op that provides the input tensor.
+  op_name: string;
+
+  // 0-based output slot index at which the op provides the input tensor.
+  output_slot: number;
+
+  // Optional recursive information about the input-providing op.
+  // This is not populated in two cases:
+  //   1. At the leaf nodes of this recursive data structure.
```

Comment:
This could be interpreted to mean that it's populated all the way down to the end of the graph.  IIUC here the "leaf nodes" are simply those for which you decided not to load the data-- in particular you load only 1 level deep, right?  Can you clarify this comment?

Reply:
Good point. Done.